doc/source/whatsnew/v1.2.0.rst

```diff
@@ -271,7 +271,8 @@ Numeric
 - Bug in :func:`to_numeric` where float precision was incorrect (:issue:`31364`)
 - Bug in :meth:`DataFrame.any` with ``axis=1`` and ``bool_only=True`` ignoring the ``bool_only`` keyword (:issue:`32432`)
 - Bug in :meth:`Series.equals` where a ``ValueError`` was raised when numpy arrays were compared to scalars (:issue:`35267`)
--
+- Bug in :meth:`DataFrame.replace` and :meth:`Series.replace` incorrectly raising ``AssertionError`` instead of ``ValueError`` when invalid parameter combinations are passed (:issue:`36045`)
+- Bug in :meth:`DataFrame.replace` and :meth:`Series.replace` with numeric values and string ``to_replace`` (:issue:`34789`)
```

Comment:
Replace is not necessarily related to "numeric"

Reply:
replace uses equality to establish what to replace, so I think at a stretch this is OK.

issue is specifically about numeric values.

if regex was involved then maybe string section, but not relevant here.

what section do you suggest?

Reply:
The message says "and string ", so the issue was not only about numeric values it seems. 

Personally I don't care that much, but *if* we are moving it, it should be an improvement, otherwise I would just leave it where it was

(in general "Numeric" is also a quite vague section)

Reply:
ahh I misread that, I thought it was replacing numeric values with strings.

will re-instate the other section and move it back.